test/rhsmlib_test/test_cloud.py

```diff
@@ -393,3 +916,67 @@ def test_conclict_in_strong_signs(self):
         self.hw_fact_collector_instance.get_all.return_value = hw_facts
         detected_clouds = detect_cloud_provider()
         self.assertEqual(detected_clouds, ['gcp', 'azure', 'aws'])
+
+    def test_collect_cloud_info_one_cloud_provider_detected(self):
+        """
+        Test the case, when we try to collect cloud info only for
+        one detected cloud provider
+        """
+        self.aws_requests_mock.get = get_only_imds_v2_is_supported
+        self.aws_requests_mock.put = put_imds_v2_token
+
+        cloud_list = ['aws']
+        cloud_info = collect_cloud_info(cloud_list)
+
+        self.assertIsNotNone(cloud_info)
+        self.assertTrue(len(cloud_info) > 0)
+        self.assertTrue('cloud_id' in cloud_info)
+        self.assertEqual(cloud_info['cloud_id'], 'aws')
+        # Test metadata
+        self.assertTrue('metadata' in cloud_info)
+        b64_metadata = cloud_info['metadata']
+        metadata = base64.b64decode(b64_metadata).decode('utf-8')
+        self.assertEqual(metadata, AWS_METADATA)
+        # Test signature
+        self.assertTrue('signature' in cloud_info)
+        b64_signature = cloud_info['signature']
+        signature = base64.b64decode(b64_signature).decode('utf-8')
+        self.assertEqual(
+            signature,
+            '-----BEGIN PKCS7-----\n' + AWS_SIGNATURE + '\n-----END PKCS7-----'
+        )
+
+    def test_collect_cloud_info_more_cloud_providers_detected(self):
+        """
+        Test the case, when we try to collect cloud info only for
+        more than one cloud providers, because more than one cloud
+        providers were detected
+        """
+        mock_result = Mock()
+        mock_result.status_code = 400
+        mock_result.text = 'Not Found 400'
+        self.azure_requests_mock.get = Mock(return_value=mock_result)
+
+        self.aws_requests_mock.get = get_only_imds_v2_is_supported
+        self.aws_requests_mock.put = put_imds_v2_token
+
+        cloud_list = ['azure', 'aws']
+        cloud_info = collect_cloud_info(cloud_list)
+
+        self.assertIsNotNone(cloud_info)
+        self.assertTrue(len(cloud_info) > 0)
+        self.assertTrue('cloud_id' in cloud_info)
+        self.assertEqual(cloud_info['cloud_id'], 'aws')
+        # Test metadata
+        self.assertTrue('metadata' in cloud_info)
+        b64_metadata = cloud_info['metadata']
+        metadata = base64.b64decode(b64_metadata).decode('utf-8')
+        self.assertEqual(metadata, AWS_METADATA)
+        # Test signature
+        self.assertTrue('signature' in cloud_info)
+        b64_signature = cloud_info['signature']
+        signature = base64.b64decode(b64_signature).decode('utf-8')
+        self.assertEqual(
+            signature,
+            '-----BEGIN PKCS7-----\n' + AWS_SIGNATURE + '\n-----END PKCS7-----'
+        )
```

Comment:
We'll have to double check but I think this will fail the style checks. I think we should add a line here.

Reply:
Fixed.